ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/ethernet_interfaces.py

```diff
@@ -0,0 +1,114 @@
+from ansible_collections.arista.avd.plugins.plugin_utils.utils import default, get
+
+from .utils import UtilsMixin
+
+
+class EthernetInterfacesMixin(UtilsMixin):
+    """
+    Mixin Class used to generate structured config for one key.
+    Class should only be used as Mixin to a AvdStructuredConfig class
+    """
+
+    @cached_property
+    def ethernet_interfaces(self) -> dict | None:
+        """
+        Return structured config for ethernet_interfaces
+        """
+        ethernet_interfaces = {}
+        for link in self._underlay_links:
+            # common values
+            ethernet_interface = {
+                "peer": get(link, "peer"),
+                "peer_interface": link["peer_interface"],
+                "peer_type": link["peer_type"],
+                "description": self._avd_interface_descriptions.underlay_ethernet_interfaces(get(link, "type"), get(link, "peer"), link["peer_interface"]),
+                "speed": get(link, "speed"),
+                "shutdown": self._shutdown_interfaces_towards_undeployed_peers and not link["peer_is_deployed"],
+            }
+
+            # L3 interface
+            if get(link, "type") == "underlay_p2p":
+                ethernet_interface.update(
+                    {
+                        "mtu": self._p2p_uplinks_mtu,
+                        "service_profile": get(self._hostvars, "p2p_uplinks_qos_profile"),
+                        "ptp": get(link, "ptp"),
+                        "mac_security": get(link, "mac_security"),
+                        "type": "routed",
+                        "ipv6_enable": get(link, "ipv6_enable"),
+                        "link_tracking_groups": get(link, "link_tracking_groups"),
+                    }
+                )
+
+                # MPLS
+                if self._mpls_lsr is True:
+                    mpls_dict = {"ip": True}
+                    if self._ldp is True:
+                        mpls_dict["ldp"] = {"interface": True, "igp_sync": True}
+
+                    ethernet_interface["mpls"] = mpls_dict
+
+                # IP address
+                if get(link, "ip_address") is not None:
+                    if "unnumbered" in link["ip_address"].lower():
+                        ethernet_interface["ip_address"] = link["ip_address"]
+                    else:
+                        ethernet_interface["ip_address"] = f"{link['ip_address']}/31"
+
+                if self._underlay_routing_protocol == "ospf":
+                    ethernet_interface["ospf_network_point_to_point"] = True
+                    ethernet_interface["ospf_area"] = self._underlay_ospf_area
+
+                if self._underlay_routing_protocol == "isis":
+                    ethernet_interface.update(
+                        {
+                            "isis_enable": self._isis_instance_name,
+                            "isis_metric": default(get(self._hostvars, "isis_default_metric"), 50),
```

Comment:
avoid using `default()` for single fallbacks. Use `get(dict, key, default=xxx)` (default is the 3rd arg and does not need kwarg, but it is easier to read this way)